Dan/README.MD

```diff
@@ -0,0 +1,2 @@
+I think the worst case time complexity is O(m*n)
```

Comment:
I think that it's a lot more than that.

You visit a block multiple times.

Reply:
Ah yah I think the worst-case is O((m-1)*(n-1)*2 + m-1 + n-1). Tho I don't think an interviewer would want that level of detail. I think saying its linear and its a little less than double m*n is sufficient

Reply:
Let's discuss this on Thursday, but I think it's a 2 ^ (m * n) solution. Generalizing it a bit, but there are m * n boxes, and you are creating a binary tree from each box (right and down). Of course, you wouldn't do it from edge columns and rows, but the complexity would be the same: 2 ^ (m * n).